kvcounter-as/assembly/index.ts

```diff
@@ -1,18 +1,27 @@
-import { handleCall, consoleLog, handleAbort } from "wapc-guest-as";
-import { Request, Response, ResponseBuilder, Handlers } from "wascc-actor-as/httpserver";
-import { Host as KV } from "wascc-actor-as/kv";
+import { Request, Response, ResponseBuilder, Handlers as HTTPHandlers } from "../../../actor-interfaces/assemblyscript/http-server/assembly/module";
+import { Host as KV } from "../../../actor-interfaces/assemblyscript/keyvalue/assembly/module";
+import { HealthCheckResponse, HealthCheckRequest, Handlers as CoreHandlers, HealthCheckResponseBuilder } from "../../../actor-interfaces/assemblyscript/actor-core/assembly/module";
 import { JSONEncoder } from "assemblyscript-json";
 
-function handleRequest(request: Request): Response {
-  const kv = new KV("");
-  const key = request.path.replaceAll("/", ":");      
-  const result = kv.atomicAdd(key, 1);  
-     
+export function wapc_init(): void {
+  CoreHandlers.registerHealthRequest(HealthCheck);
+  HTTPHandlers.registerHandleRequest(HandleRequest);
+}
+
+function HealthCheck(request: HealthCheckRequest): HealthCheckResponse {
+  return new HealthCheckResponseBuilder().withHealthy(true).withMessage("AssemblyScript KVCounter Healthy").build();
+}
+
+function HandleRequest(request: Request): Response {
+  const kv = new KV("default");
```

Comment:
does AssemblyScript support default values? If so, then maybe we should file an issue with the generator templates to default the namespace to `default`

Reply:
One thing we can do is change the provider constructor to:
```
  constructor(binding: string = "default") {
    this.binding = binding;
  }
```
Which is a typescript construct, providing this value allows you to give no argument to the contstructor (e.g. initialize like `const kv = new KV();`, and if a value is provided then it's assigned